### DISCUSSION TITLE
CALCULADORA C#

### DISCUSSION THREAD

**Discussion by ycaleb** [balta-io/forum#66] [Tira Dúvidas] [answered]
Boa tarde, na calculadora de fundamentos c#, meu codigo está igual ao codigo fonte feito pelo Balta, porém está aparecendo aviso de exção em todas as ReadLines.
![image](https://user-images.githubusercontent.com/111933424/197411489-8e85282d-9f9c-4ef7-bb53-dad1c5325abc.png)
![image](https://user-images.githubusercontent.com/111933424/197411495-2cbc06ba-8f7a-407e-bc81-c535393bb44e.png)

**Comment by andrebaltieri** [comment, marked as answer]
Boa noite, @ycaleb, como vai?

Este aviso (Não chega a ser um erro, sua aplicação compila) te diz que você tem variáveis que não são **nullable** e também não foram inicializadas.

Para corrigir este problema você pode:
```csharp
string minhaVariavel = string.Empty; // Inicializa a variável
string? minhaVariavel; // Declara a variável como nullable (Que pode ser nula)
string minhaVariavel = null!; // Diz que a variável não é nullable, não foi inicializada, mas que você garante que irá ter um valor nela futuramente
```

**Reply by ycaleb** [reply]
Boa tarde Andre, espero que esteja tudo tranquilo.

Estou começando agora então estou tendo um pouco de dificuldade, sabe me dizer em que local do código eu declaro essa variável? Nao estou sabendo direito como aplicar esse código que voce mandou...
Desde ja, agradeço.

**Comment by andrebaltieri** [comment]
@ycaleb se está começando (Curso de Fundamentos) então fica tranquilo, só ignora estas mensagens que depois você vai ver mais sobre isto <3
